botocore/retryhandler.py

```diff
@@ -30,8 +30,8 @@
 # this mapping with more specific exceptions.
 EXCEPTION_MAP = {
     'GENERAL_CONNECTION_ERROR': [
-        ConnectionError, ClosedPoolError, Timeout,
-        EndpointConnectionError
+        ReadTimeoutError, ProxyConnectionError, EndpointConnectionError,
```

Comment:
Can we just use `HTTPClientError` here?

Reply:
Possibly, I was thinking it might be nice to just have some of the `HTTPClientError` exception types extend from a `RetryableHTTPClientError` and then use that instead.